src/Val.cc

```diff
@@ -524,102 +536,125 @@ static ZeekJson BuildJSON(Val* val, bool only_loggable=false, RE_Matcher* re=nul
 				else
 					entry_key = lv->Ref();
 
-				ZeekJson key_json = BuildJSON(entry_key, only_loggable, re);
-
 				if ( tval->Type()->IsSet() )
-					j.emplace_back(std::move(key_json));
+					BuildJSON(writer, entry_key, only_loggable, re);
 				else
 					{
 					Val* entry_value = entry->Value();
 
-					string key_string;
-					if ( key_json.is_string() )
-						key_string = key_json;
-					else
-						key_string = key_json.dump();
+					rapidjson::StringBuffer buffer;
+					NullDoubleWriter key_writer(buffer);
+					BuildJSON(key_writer, entry_key, only_loggable, re);
+					string key_str = buffer.GetString();
+					if ( key_str[0] == '"')
+						key_str = key_str.substr(1);
+					if ( key_str[key_str.length()-1] == '"')
```

Comment:
These two `if` seem a bit odd: Are there quotes added to the sub value or not? Can it happen that there's a quote at the front, but not the back (or vice versa)?

Reply:
It's possible that we can get either a string (which will have quotes) or some other non-string value (which won't) back from `GetString()`. In the case of a string, we need to remove the quotes because otherwise we end up with something like `"\"string\""` in the output.

Reply:
As for whether there will ever be a quote in the front and not the back, probably not.

Reply:
ok, then I think I'll change it it to ``if ( key_str[0] == '"' && key_str[key_str.length()-1] == '"' ) ...`` (oh, and a size check for the -1 I guess)